### PR TITLE
Anton/contribute create dataset

### PR DESCRIPTION
**Pull request by antondelchev**
Signed-off-by: Anton Delchev <dlchv.mail@gmail.com>

### CODE REVIEW

File: src/zos_files/zowe/zos_files_for_zowe_sdk/files.py [hunk 1]
```diff
@@ -156,6 +158,10 @@ def create_data_set(self, dataset_name, options = {}):
         json
         """
 
+        if options.get("like") is None:
+            if options.get("primary") is None and options.get("lrecl") is None:
```
Comment (t1m0thyj) [inline src/zos_files/zowe/zos_files_for_zowe_sdk/files.py:162]: I believe this `and` should be an `or`, to throw an error when just one option is missing.

**Review by t1m0thyj — COMMENTED** [review]
Thank you for adding the `create_default_data_set` method, I think it will be very useful 👍 

Left a comment 🙂 Could you please also update this PR with the "next" branch to resolve conflicts?

File: src/zos_files/zowe/zos_files_for_zowe_sdk/files.py (hunk 1 above)
Comment (antondelchev) [reply src/zos_files/zowe/zos_files_for_zowe_sdk/files.py:162]: Thank you for the feedback Timothy, I'll make the necessary changes!

File: src/zos_files/zowe/zos_files_for_zowe_sdk/files.py [hunk 2]
```diff
@@ -217,6 +223,90 @@ def create_data_set(self, dataset_name, options = {}):
...
+                "blksize": 27998,
+                "lrecl": 27998,
+                "dirblk": 25
+            }
+        elif default_type == "vsam":
+            options = {
+                "dsorg": "INDEXED",
+                "alcunit": "KB",
+                "primary": 840
+            }
```
Comment (t1m0thyj) [inline src/zos_files/zowe/zos_files_for_zowe_sdk/files.py:301]: VSAM is a special data set type that cannot be created with the same API as other types.

To create a VSAM, you need to use the "Access Method Services Interface" documented [here](https://www.ibm.com/docs/en/zos/2.4.0?topic=interface-access-method-services). This is added scope outside of the issue addressed in this PR, so I'd suggest removing VSAM for now and we can handle it as a separate issue in the future 🙂

Comment (antondelchev) [reply src/zos_files/zowe/zos_files_for_zowe_sdk/files.py:301]: Alright, thank you, I'll edit this!

File: src/zos_files/zowe/zos_files_for_zowe_sdk/files.py [hunk 3]
```diff
@@ -163,6 +165,10 @@ def create_data_set(self, dataset_name, options = {}):
         json
         """
 
+        if options.get("like") is None:
+            if options.get("primary") is None or options.get("lrecl") is None:
+                raise KeyError
+
         for opt in ("volser", "unit", "dsorg", "alcunit",
```
Comment (zFernand0) [inline src/zos_files/zowe/zos_files_for_zowe_sdk/files.py:172]: I believe that whenever the `like` option is specified in the `dict`, we shouldn't be adding any default values (e.g. `alcunit: TRK`, `recfm: F`)
Maybe we just need to include this `for` loop inside the `if('like') is None` condition 😋

**Review by zFernand0 — CHANGES_REQUESTED** [review]
I agree that `create_default_data_set` is super useful! 🥳 
I do have one comment (and thanks @t1m0thyj for helping me understand what was happening)

I left a comment which I hope makes sense 😋

File: src/zos_files/zowe/zos_files_for_zowe_sdk/files.py (hunk 3 above)
Comment (antondelchev) [reply src/zos_files/zowe/zos_files_for_zowe_sdk/files.py:172]: Thank you, I will change that!

**Review by t1m0thyj — APPROVED** [review]

**Review by t1m0thyj — CHANGES_REQUESTED** [review]
Please fix the failing unit test 🙂

**Review by t1m0thyj — APPROVED** [review]

**Pull request merged**